src/coq_rules.ml

```diff
@@ -112,14 +112,11 @@ let create_ccoq sctx ~dir =
   }
 
 (* get_libraries from Coq's ML dependencies *)
-let libs_of_coq_deps ~loc ~scope libs =
-  let lib_db = Scope.libs scope in
-  List.concat_map ~f:Dune_file.Lib_dep.to_lib_names libs
-  |> Lib.DB.find_many ~loc lib_db
-  |> Result.ok_exn
+let libs_of_coq_deps ~lib_db libs =
+  Result.List.map ~f:(Lib.DB.resolve lib_db) libs |> Result.ok_exn
```

Comment:
I think you should be able to use find_many here, no?

Reply:
I would less worse error messages in that case, right? `find_many` doesn't seem to track individual `loc`, does it?

Reply:
Ah yeah, I suppose I had something closer to `resolve_user_written_deps_for_exes` which takes a list of libraries and locs. The goal here is to have the library database be responsible for the error handling. Here you have the coq rules specifying that no lookup will be done after the first lib fails to resolve. IMHO, that should be a property of the db itself.

In any case, this isn't so important to solve now. I'm still forming the vision for how a generalized library database is going to look like, and this kind of error handling is something that I'd like this new abstraction to take care of.

Reply:
Sure we could a higher-level resolve rule, however `find_many` indeed doesn't seem to have the right semantics if you want accurate error reporting now; that's what I meant.